lib/charms/opensearch/v0/opensearch_nodes_exclusions.py

```diff
@@ -70,55 +72,134 @@ def delete_current(self) -> None:
                 self._scope, ALLOCS_TO_DELETE, ",".join(current_allocations)
             )
 
+    def _removed_units_to_cleanup(self) -> Optional[List[str]]:
+        """Deletes all units that have left the cluster via Juju.
+
+        This method ensures we keep a small list of voting exclusions at all times.
+        """
+        deployment_desc = self._charm.opensearch_peer_cm.deployment_desc()
+        if not deployment_desc or deployment_desc.typ not in [
+            DeploymentType.MAIN_ORCHESTRATOR,
+            DeploymentType.FAILOVER_ORCHESTRATOR,
```

Comment:
We should only do this for the `MAIN_ORCHESTRATOR` - as it's the only one supposed to be at the center of the star.